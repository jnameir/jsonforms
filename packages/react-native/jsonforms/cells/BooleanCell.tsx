import {
    CellProps,
    isBooleanControl,
    RankedTester,
    rankWith,
} from "@jsonforms/core";
import { withJsonFormsCellProps } from "@jsonforms/react";
import React from "react";
import { Checkbox } from "react-native-material-ui";

export const BooleanCell = (props: CellProps) => {
    const { data, enabled, path, handleChange } = props;

    const onChange = () => {
        handleChange(path, !data);
    };

    return (
        <Checkbox
            label=""
            value={path}
            checked={!!data}
            onCheck={onChange}
            disabled={!enabled}
        />
    );
};

/**
 * Default tester for boolean controls.
 * @type {RankedTester}
 */
export const booleanCellTester: RankedTester = rankWith(2, isBooleanControl);

export default withJsonFormsCellProps(BooleanCell);
