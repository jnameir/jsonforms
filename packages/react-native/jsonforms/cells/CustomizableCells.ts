export { BooleanCell } from "./BooleanCell";
export { DateCell } from "./DateCell";
export { DateTimeCell } from "./DateTimeCell";
export { EnumCell } from "./EnumCell";
export { MultiEnumCell } from "./MultiEnumCell";
export { IntegerCell } from "./IntegerCell";
export { NumberCell } from "./NumberCell";
export { NumberFormatCell } from "./NumberFormatCell";
export { TextCell } from "./TextCell";
export { TextAreaCell } from "./TextAreaCell";
export { TimeCell } from "./TimeCell";
