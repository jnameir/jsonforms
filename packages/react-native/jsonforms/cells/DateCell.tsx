import {
    CellProps,
    isDateControl,
    RankedTester,
    rankWith,
} from "@jsonforms/core";
import { withJsonFormsCellProps } from "@jsonforms/react";
import React from "react";
import { DatePickerAndroid } from "react-native";
import { TextField } from "react-native-material-textfield";
import { Button } from "react-native-material-ui";
import { twoDigits } from "../util";

export const DateCell = (props: CellProps) => {
    const { data, path, handleChange } = props;
    const formattedDate = data
        ? `${data.split("-")[2]}.${data.split("-")[1]}.${data.split("-")[0]}`
        : "";

    const onChange = async () => {
        try {
            const date = await DatePickerAndroid.open({});

            if (date.action !== DatePickerAndroid.dismissedAction) {
                handleChange(
                    path,
                    `${date.year}-${twoDigits(date.month + 1)}-${twoDigits(
                        date.day,
                    )}`,
                );
            }
        } catch ({ code, message }) {
            console.log("wrong date format");
        }
    };

    return (
        <>
            <TextField value={formattedDate} editable={false} />
            <Button onPress={onChange} text="Pick date" />
        </>
    );
};

/**
 * Default tester for date controls.
 * @type {RankedTester}
 */
export const dateCellTester: RankedTester = rankWith(2, isDateControl);

export default withJsonFormsCellProps(DateCell);
