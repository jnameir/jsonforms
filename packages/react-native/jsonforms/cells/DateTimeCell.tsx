import {
    CellProps,
    isDateTimeControl,
    RankedTester,
    rankWith,
} from "@jsonforms/core";
import { withJsonFormsCellProps } from "@jsonforms/react";
import DateTimePicker from "@react-native-community/datetimepicker";
import React, { useState } from "react";
import { TextField } from "react-native-material-textfield";
import { Button } from "react-native-material-ui";

export const DateTimeCell = (props: CellProps) => {
    const { data, handleChange, path } = props;

    const [date, setDate] = useState(data ? new Date(data) : new Date());
    const [visible, setVisible] = useState(false);
    const [mode, setMode] = useState<"date" | "time">("date");

    const onChange = (event, selectedDate) => {
        setVisible(false);
        const currentDate = selectedDate || date;
        handleChange(path, currentDate.toISOString());
        setDate(currentDate);
    };

    const showDatepicker = () => {
        setVisible(true);
        setMode("date");
    };

    const showTimepicker = () => {
        setVisible(true);
        setMode("time");
    };

    return (
        <>
            {visible && (
                <DateTimePicker
                    value={date}
                    mode={mode}
                    is24Hour={true}
                    display="default"
                    onChange={onChange}
                />
            )}
            <TextField value={date.toLocaleString()} />
            <Button onPress={showDatepicker} text="Pick date" />
            <Button onPress={showTimepicker} text="Pick time" />
        </>
    );
};

/**
 * Default tester for datetime controls.
 * @type {RankedTester}
 */
export const dateTimeCellTester: RankedTester = rankWith(2, isDateTimeControl);

export default withJsonFormsCellProps(DateTimeCell);
