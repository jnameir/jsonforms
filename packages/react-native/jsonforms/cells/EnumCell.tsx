import {
    EnumCellProps,
    isEnumControl,
    RankedTester,
    rankWith
} from "@jsonforms/core";
import { withJsonFormsEnumCellProps } from "@jsonforms/react";
import { Picker } from "@react-native-picker/picker";
import React from "react";
import { StyleSheet, View } from "react-native";

export const EnumCell = (props: EnumCellProps) => {
    const { data, enabled, path, handleChange, options, visible } = props;


    const onChange = (value) => {
        handleChange(path, value);
    };

    const showBorder = visible ? { style: styles.contentBorder } : null;

    return (
        <View style={styles.container}>
            <View {...showBorder}>
                <Picker
                    enabled={enabled}
                    selectedValue={data || ""}
                    onValueChange={onChange}
                >
                    {[
                        <Picker.Item
                            label="Bitte auswÃ¤hlen"
                            value=""
                            key="empty"
                        />,
                    ].concat(
                        options.map((optionValue) => (
                            <Picker.Item
                                value={optionValue.value}
                                label={optionValue.label}
                                key={optionValue.value}
                            />
                        )),
                    )}
                </Picker>
            </View>
        </View>
    );
};

/**
 * Default tester for enum controls.
 * @type {RankedTester}
 */
export const enumCellTester: RankedTester = rankWith(2, isEnumControl);

export default withJsonFormsEnumCellProps(EnumCell);

const styles = StyleSheet.create({
    container: {
        width: "30%",
        margin: 5,
    },
    contentBorder: {
        borderWidth: 1,
        borderColor: "darkgrey",
    },
});
