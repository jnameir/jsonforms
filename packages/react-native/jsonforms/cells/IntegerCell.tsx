import {
    CellProps,
    isIntegerControl,
    RankedTester,
    rankWith,
} from "@jsonforms/core";
import { withJsonFormsCellProps } from "@jsonforms/react";
import merge from "lodash/merge";
import React from "react";
import { TextField } from "react-native-material-textfield";

export const IntegerCell = (props: CellProps) => {
    const { config, data, enabled, uischema, schema, path, handleChange } =
        props;

    const onChange = (value: string) => {
        handleChange(path, parseInt(value));
    };

    const { maxLength } = schema;
    const appliedUiSchemaOptions = merge({}, config, uischema.options);

    return (
        <TextField
            value={data?.toString() || ""}
            onChangeText={onChange}
            editable={enabled}
            autoFocus={appliedUiSchemaOptions.focus}
            maxLength={appliedUiSchemaOptions.restrict ? maxLength : undefined}
        />
    );
};

/**
 * Default tester for integer controls.
 * @type {RankedTester}
 */
export const integerCellTester: RankedTester = rankWith(2, isIntegerControl);

export default withJsonFormsCellProps(IntegerCell);
