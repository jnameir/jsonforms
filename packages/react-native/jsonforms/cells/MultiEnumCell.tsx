import {
    and,
    EnumCellProps,
    enumToEnumOptionMapper,
    RankedTester,
    rankWith,
    schemaMatches,
    schemaTypeIs,
    uiTypeIs
} from "@jsonforms/core";
import { withJsonFormsEnumCellProps } from "@jsonforms/react";
import React from "react";
import { StyleSheet, View } from "react-native";
import SectionedMultiSelect from "react-native-sectioned-multi-select";
import Icons from "react-native-vector-icons/MaterialIcons";


export const MultiEnumCell = (props: Partial<EnumCellProps>) => {
    const { data, path, schema, handleChange, visible, enabled, errors } =
        props;

    const onChange = (value) => {
        handleChange(path, value?.length > 0 ? value : undefined);
    };

    let items = [];

    if (schema.items && "enum" in schema.items) {
        items = schema.items.enum;
    } else {
        return <></>;
    }

    const options = items.map((e) => enumToEnumOptionMapper(e));
    const preparedOptions = [{ id: 0, name: schema.title, children: [] }];

    options.forEach((e) =>
        preparedOptions[0].children.push({
            id: e.value,
            name: e.label,
        }),
    );

    const showBorder = visible
        ? errors
            ? { style: { ...styles.errorStyle } }
            : { style: { ...styles.contentBorder } }
        : null;

    /** workaround for fixing the wrong type definition of the 3rd party package react-native-sectioned-multi-select */
    const Icon = Icons as unknown as JSX.Element;

    return (
        <View style={styles.container}>
            <View {...showBorder} data-testid="MultiEnumCell-container">
                <SectionedMultiSelect
                    icons={multiSelectStyles.icons}
                    IconRenderer={Icon}
                    uniqueKey="id"
                    subKey="children"
                    items={preparedOptions}
                    selectText="Bitte auswählen..."
                    showDropDowns={false}
                    readOnlyHeadings={true}
                    hideSearch={false}
                    searchPlaceholderText="Suche"
                    confirmText="Übernehmen"
                    selectedText="ausgewählt"
                    selectedItems={data || []}
                    onSelectedItemsChange={onChange}
                    disabled={!enabled}
                    styles={multiSelectStyles.styles}
                />
            </View>
        </View>
    );
};
/**
 * Default tester for multi select enum controls.
 * @type {RankedTester}
 */

const isMultiEnumControl = and(
    uiTypeIs("Control"),
    schemaTypeIs("array"),
    schemaMatches(
        (schema) =>
            Object.prototype.hasOwnProperty.call(schema, "items") &&
            Object.prototype.hasOwnProperty.call(schema.items, "enum"),
    ),
);

export const multiEnumCellTester: RankedTester = rankWith(
    2,
    isMultiEnumControl,
);

export default withJsonFormsEnumCellProps(MultiEnumCell);

const styles = StyleSheet.create({
    container: {
        width: "50%",
        margin: 5,
    },
    contentBorder: {
        borderWidth: 1,
        borderColor: "darkgrey",
    },
    errorStyle: {
        borderWidth: 1,
        borderColor: "red",
    },
});

const multiSelectStyles = {
    icons: {
        search: {
            name: "search", // search input
            size: 24,
        },
        arrowUp: {
            name: "keyboard-arrow-up", // dropdown toggle
            size: 22,
        },
        arrowDown: {
            name: "keyboard-arrow-down", // dropdown toggle
            size: 22,
        },
        selectArrowDown: {
            name: "keyboard-arrow-down", // select
            size: 24,
        },
        close: {
            name: "close", // chip close
            size: 16,
        },
        check: {
            name: "check", // selected item
            size: 16,
        },
        cancel: {
            name: "cancel", // cancel button
            size: 18,
        },
    },
    styles: {
        selectedItem: {
            backgroundColor: "rgba(0,0,0,0.1)",
        },
        selectedSubItem: {
            backgroundColor: "rgba(0,0,0,0.1)",
        },
        button: {
            backgroundColor: "green",
        },
    },
};
