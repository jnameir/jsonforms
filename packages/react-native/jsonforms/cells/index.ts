import BooleanCell, { booleanCellTester } from "./BooleanCell";
import * as Customizable from "./CustomizableCells";
import DateCell, { dateCellTester } from "./DateCell";
import DateTimeCell, { dateTimeCellTester } from "./DateTimeCell";
import EnumCell, { enumCellTester } from "./EnumCell";
import MultiEnumCell, { multiEnumCellTester } from "./MultiEnumCell";
import IntegerCell, { integerCellTester } from "./IntegerCell";
import NumberCell, { numberCellTester } from "./NumberCell";
import NumberFormatCell, { numberFormatCellTester } from "./NumberFormatCell";
import TextAreaCell, { textAreaCellTester } from "./TextAreaCell";
import TextCell, { textCellTester } from "./TextCell";
import TimeCell, { timeCellTester } from "./TimeCell";

export {
    BooleanCell,
    booleanCellTester,
    DateCell,
    dateCellTester,
    DateTimeCell,
    dateTimeCellTester,
    EnumCell,
    enumCellTester,
    MultiEnumCell,
    multiEnumCellTester,
    IntegerCell,
    integerCellTester,
    NumberCell,
    numberCellTester,
    NumberFormatCell,
    numberFormatCellTester,
    TextCell,
    textCellTester,
    TextAreaCell,
    textAreaCellTester,
    TimeCell,
    timeCellTester,
};
export { Customizable };
