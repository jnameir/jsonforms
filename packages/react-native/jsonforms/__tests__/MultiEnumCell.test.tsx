import { render } from "@entyre/test-utilities";
import { EnumCellProps } from "@jsonforms/core";
import React from "react";
import { View } from "react-native";
import SectionedMultiSelect from "react-native-sectioned-multi-select";
import { MultiEnumCell } from "../../jsonforms/cells/MultiEnumCell";

const props: Partial<EnumCellProps> = {
    data: "",
    path: "#/test",
    schema: {
        items: {
            type: "string",
            enum: ["option1", "option2", "option3"],
        },
    },
    visible: true,
    enabled: true,
};

function renderCell(properties: Partial<EnumCellProps>) {
    const root = render<typeof MultiEnumCell>(() => (
        <MultiEnumCell {...properties} />
    ));

    const { instance } = root;

    return {
        root,
        instance,
    };
}

describe("JSONForms: MultiEnumCell", () => {
    test("success snapshot neutral", () => {
        const handleChange = jest.fn();
        props.handleChange = handleChange;
        const { root } = renderCell(props);
        expect(root.json).toMatchSnapshot();
    });

    test("success select items", () => {
        const handleChange = jest.fn();
        props.handleChange = handleChange;
        const { instance } = renderCell(props);
        const selectedValues = ["option1", "option2"];
        const selectables = instance.findAllByType(SectionedMultiSelect);
        selectables[0].props.onSelectedItemsChange(selectedValues);
        expect(handleChange).toHaveBeenCalledWith(props.path, selectedValues);
    });

    test("success selected items", () => {
        const selectedValues = ["option1", "option2"];
        props.data = selectedValues;
        const { instance } = renderCell(props);
        const selectables = instance.findAllByType(SectionedMultiSelect);
        expect(selectables[0].props.selectedItems).toEqual(selectedValues);
    });

    test("success not visible", () => {
        props.visible = false;
        const { instance } = renderCell(props);
        const selectables = instance.findAllByType(View);
        expect(selectables[1].props.style).toEqual(undefined);
    });

    test("error rendering cell", () => {
        props.schema = {};
        const { instance } = renderCell(props);
        expect(instance).toEqual(undefined);
    });
});
