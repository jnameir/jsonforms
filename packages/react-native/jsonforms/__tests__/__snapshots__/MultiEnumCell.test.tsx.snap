// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JSONForms: MultiEnumCell success snapshot neutral 1`] = `
<View
  style={
    Object {
      "margin": 5,
      "width": "50%",
    }
  }
>
  <View
    data-testid="MultiEnumCell-container"
    style={
      Object {
        "borderColor": "darkgrey",
        "borderWidth": 1,
      }
    }
  >
    <View>
      <Modal
        animationType="fade"
        hardwareAccelerated={false}
        onRequestClose={[Function]}
        supportedOrientations={
          Array [
            "portrait",
            "landscape",
          ]
        }
        transparent={true}
        visible={false}
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "rgba(0,0,0,0.5)",
                "flex": 1,
              },
              Object {},
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "bottom": 0,
                  "flex": 1,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "zIndex": 0,
                },
                undefined,
              ]
            }
          />
          <View
            style={
              Array [
                Object {
                  "alignSelf": "stretch",
                  "backgroundColor": "white",
                  "borderRadius": 6,
                  "flex": 1,
                  "marginHorizontal": 18,
                  "marginVertical": 26,
                  "overflow": "hidden",
                },
                Object {},
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "flexDirection": "row",
                    "paddingVertical": 5,
                  },
                  Object {
                    "backgroundColor": "#f8f8f8",
                    "flexDirection": "row",
                  },
                ]
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "justifyContent": "center",
                  }
                }
              >
                <Text />
              </View>
              <TextInput
                allowFontScaling={true}
                autoFocus={false}
                onChangeText={[Function]}
                placeholder="Suche"
                placeholderTextColor="#999"
                rejectResponderTermination={true}
                selectTextOnFocus={true}
                selectionColor="rgba(0,0,0,0.2)"
                style={
                  Array [
                    Object {
                      "flex": 1,
                      "fontSize": 17,
                      "paddingVertical": 8,
                    },
                    Object {
                      "fontFamily": "Avenir",
                      "fontWeight": "200",
                    },
                    Object {},
                  ]
                }
                underlineColorAndroid="transparent"
                value=""
              />
            </View>
            <View
              keyboardShouldPersistTaps="always"
              style={
                Array [
                  Object {
                    "flex": 1,
                    "paddingHorizontal": 12,
                  },
                  Object {},
                ]
              }
            >
              <View>
                <View>
                  <View>
                    <RCTScrollView
                      ItemSeparatorComponent={[Function]}
                      ListFooterComponent={[Function]}
                      data={
                        Array [
                          Object {
                            "children": Array [
                              Object {
                                "id": "option1",
                                "name": "option1",
                              },
                              Object {
                                "id": "option2",
                                "name": "option2",
                              },
                              Object {
                                "id": "option3",
                                "name": "option3",
                              },
                            ],
                            "id": 0,
                            "name": undefined,
                          },
                        ]
                      }
                      disableVirtualization={false}
                      extraData={Array []}
                      getItem={[Function]}
                      getItemCount={[Function]}
                      horizontal={false}
                      initialNumToRender={15}
                      keyExtractor={[Function]}
                      keyboardShouldPersistTaps="always"
                      maxToRenderPerBatch={10}
                      onContentSizeChange={[Function]}
                      onEndReachedThreshold={2}
                      onLayout={[Function]}
                      onMomentumScrollEnd={[Function]}
                      onScroll={[Function]}
                      onScrollBeginDrag={[Function]}
                      onScrollEndDrag={[Function]}
                      removeClippedSubviews={true}
                      renderItem={[Function]}
                      scrollEventThrottle={50}
                      stickyHeaderIndices={Array []}
                      updateCellsBatchingPeriod={50}
                      viewabilityConfigCallbackPairs={Array []}
                      windowSize={21}
                    >
                      <View>
                        <View
                          onLayout={[Function]}
                          style={null}
                        >
                          <View>
                            <View>
                              <View
                                style={
                                  Object {
                                    "backgroundColor": "#fff",
                                    "flex": 1,
                                    "flexDirection": "row",
                                  }
                                }
                              >
                                <View
                                  accessible={true}
                                  focusable={true}
                                  onClick={[Function]}
                                  onResponderGrant={[Function]}
                                  onResponderMove={[Function]}
                                  onResponderRelease={[Function]}
                                  onResponderTerminate={[Function]}
                                  onResponderTerminationRequest={[Function]}
                                  onStartShouldSetResponder={[Function]}
                                  style={
                                    Object {
                                      "alignItems": "center",
                                      "flex": 1,
                                      "flexDirection": "row",
                                      "justifyContent": "flex-start",
                                      "opacity": undefined,
                                      "paddingVertical": 6,
                                    }
                                  }
                                >
                                  <Text
                                    numberOfLines={null}
                                    style={
                                      Array [
                                        Object {
                                          "color": "#2e2e2e",
                                          "flex": 1,
                                        },
                                        Object {
                                          "fontFamily": "Avenir",
                                          "fontWeight": "bold",
                                        },
                                        Object {
                                          "fontSize": 17,
                                        },
                                        false,
                                      ]
                                    }
                                  />
                                </View>
                              </View>
                              <View
                                ItemSeparatorComponent={[Function]}
                                data={
                                  Array [
                                    Object {
                                      "id": "option1",
                                      "name": "option1",
                                    },
                                    Object {
                                      "id": "option2",
                                      "name": "option2",
                                    },
                                    Object {
                                      "id": "option3",
                                      "name": "option3",
                                    },
                                  ]
                                }
                                disableVirtualization={false}
                                extraData={Array []}
                                getItem={[Function]}
                                getItemCount={[Function]}
                                horizontal={false}
                                initialNumToRender={20}
                                keyExtractor={[Function]}
                                maxToRenderPerBatch={10}
                                onContentSizeChange={[Function]}
                                onEndReachedThreshold={2}
                                onLayout={[Function]}
                                onMomentumScrollEnd={[Function]}
                                onScroll={[Function]}
                                onScrollBeginDrag={[Function]}
                                onScrollEndDrag={[Function]}
                                removeClippedSubviews={false}
                                renderItem={[Function]}
                                scrollEventThrottle={50}
                                stickyHeaderIndices={Array []}
                                updateCellsBatchingPeriod={50}
                                viewabilityConfigCallbackPairs={Array []}
                                windowSize={21}
                              >
                                <View
                                  onLayout={[Function]}
                                  style={null}
                                >
                                  <View
                                    style={
                                      Object {
                                        "backgroundColor": "#ffffff",
                                        "flex": 1,
                                        "flexDirection": "row",
                                      }
                                    }
                                  >
                                    <View
                                      accessible={true}
                                      focusable={true}
                                      onClick={[Function]}
                                      onResponderGrant={[Function]}
                                      onResponderMove={[Function]}
                                      onResponderRelease={[Function]}
                                      onResponderTerminate={[Function]}
                                      onResponderTerminationRequest={[Function]}
                                      onStartShouldSetResponder={[Function]}
                                      style={
                                        Object {
                                          "alignItems": "center",
                                          "flex": 1,
                                          "flexDirection": "row",
                                          "justifyContent": "flex-start",
                                          "opacity": undefined,
                                          "paddingVertical": 6,
                                        }
                                      }
                                    >
                                      <Text
                                        numberOfLines={null}
                                        style={
                                          Array [
                                            Object {
                                              "color": "#848787",
                                              "flex": 1,
                                            },
                                            Object {
                                              "fontFamily": "Avenir",
                                              "fontWeight": "200",
                                            },
                                            Object {
                                              "fontSize": 15,
                                              "paddingLeft": 8,
                                            },
                                            false,
                                          ]
                                        }
                                      >
                                        option1
                                      </Text>
                                    </View>
                                  </View>
                                  <View
                                    style={
                                      Array [
                                        Object {
                                          "alignSelf": "stretch",
                                          "backgroundColor": "#dadada",
                                          "flex": 1,
                                          "height": 0.5,
                                        },
                                        Object {
                                          "height": 0,
                                        },
                                      ]
                                    }
                                  />
                                </View>
                                <View
                                  onLayout={[Function]}
                                  style={null}
                                >
                                  <View
                                    style={
                                      Object {
                                        "backgroundColor": "#ffffff",
                                        "flex": 1,
                                        "flexDirection": "row",
                                      }
                                    }
                                  >
                                    <View
                                      accessible={true}
                                      focusable={true}
                                      onClick={[Function]}
                                      onResponderGrant={[Function]}
                                      onResponderMove={[Function]}
                                      onResponderRelease={[Function]}
                                      onResponderTerminate={[Function]}
                                      onResponderTerminationRequest={[Function]}
                                      onStartShouldSetResponder={[Function]}
                                      style={
                                        Object {
                                          "alignItems": "center",
                                          "flex": 1,
                                          "flexDirection": "row",
                                          "justifyContent": "flex-start",
                                          "opacity": undefined,
                                          "paddingVertical": 6,
                                        }
                                      }
                                    >
                                      <Text
                                        numberOfLines={null}
                                        style={
                                          Array [
                                            Object {
                                              "color": "#848787",
                                              "flex": 1,
                                            },
                                            Object {
                                              "fontFamily": "Avenir",
                                              "fontWeight": "200",
                                            },
                                            Object {
                                              "fontSize": 15,
                                              "paddingLeft": 8,
                                            },
                                            false,
                                          ]
                                        }
                                      >
                                        option2
                                      </Text>
                                    </View>
                                  </View>
                                  <View
                                    style={
                                      Array [
                                        Object {
                                          "alignSelf": "stretch",
                                          "backgroundColor": "#dadada",
                                          "flex": 1,
                                          "height": 0.5,
                                        },
                                        Object {
                                          "height": 0,
                                        },
                                      ]
                                    }
                                  />
                                </View>
                                <View
                                  onLayout={[Function]}
                                  style={null}
                                >
                                  <View
                                    style={
                                      Object {
                                        "backgroundColor": "#ffffff",
                                        "flex": 1,
                                        "flexDirection": "row",
                                      }
                                    }
                                  >
                                    <View
                                      accessible={true}
                                      focusable={true}
                                      onClick={[Function]}
                                      onResponderGrant={[Function]}
                                      onResponderMove={[Function]}
                                      onResponderRelease={[Function]}
                                      onResponderTerminate={[Function]}
                                      onResponderTerminationRequest={[Function]}
                                      onStartShouldSetResponder={[Function]}
                                      style={
                                        Object {
                                          "alignItems": "center",
                                          "flex": 1,
                                          "flexDirection": "row",
                                          "justifyContent": "flex-start",
                                          "opacity": undefined,
                                          "paddingVertical": 6,
                                        }
                                      }
                                    >
                                      <Text
                                        numberOfLines={null}
                                        style={
                                          Array [
                                            Object {
                                              "color": "#848787",
                                              "flex": 1,
                                            },
                                            Object {
                                              "fontFamily": "Avenir",
                                              "fontWeight": "200",
                                            },
                                            Object {
                                              "fontSize": 15,
                                              "paddingLeft": 8,
                                            },
                                            false,
                                          ]
                                        }
                                      >
                                        option3
                                      </Text>
                                    </View>
                                  </View>
                                </View>
                              </View>
                            </View>
                          </View>
                        </View>
                        <View
                          onLayout={[Function]}
                        >
                          <View />
                        </View>
                      </View>
                    </RCTScrollView>
                  </View>
                </View>
              </View>
            </View>
            <View
              style={
                Object {
                  "flexDirection": "row",
                }
              }
            >
              <View
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "flex": 1,
                    "opacity": undefined,
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "#3f51b5",
                        "borderRadius": 0,
                        "flex": 0,
                        "flexDirection": "row",
                        "justifyContent": "center",
                        "paddingHorizontal": 16,
                        "paddingVertical": 8,
                      },
                      Object {
                        "backgroundColor": "green",
                      },
                    ]
                  }
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "#ffffff",
                          "fontSize": 18,
                        },
                        Object {
                          "fontFamily": "Avenir",
                          "fontWeight": "bold",
                        },
                        Object {},
                      ]
                    }
                  >
                    Übernehmen
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </Modal>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "flexWrap": "wrap",
            },
            Object {
              "borderRadius": 4,
              "marginBottom": 15,
              "marginTop": 5,
              "paddingHorizontal": 10,
              "paddingVertical": 12,
            },
          ]
        }
      >
        <Text
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "#333",
                "flex": 1,
                "fontSize": 16,
              },
              Object {},
            ]
          }
        >
          Bitte auswählen...
        </Text>
        <Text />
      </View>
    </View>
  </View>
</View>
`;
